### DISCUSSION TITLE
How to Convert Attribute to Vec<u8>

### DISCUSSION THREAD

**Discussion by shivam560** [parallaxsecond/rust-cryptoki#136] [Q&A] [answered]
Hey,
I am trying to get the value of Public key and I need to generate the CSR using these keys. But in order to generate the CSR using openssl we need to have a key of type PKEY . Please refer the below code.

```rust
pub fn create_key_pair() -> Result<(),()> {
    let (pkcs111,slit) = init_pins();
    let session = pkcs111.open_rw_session(slit).unwrap();
    session.login(cryptoki::session::UserType::User, Some("1234"));
    let class = ObjectClass::PUBLIC_KEY;
    let pub_template =  Attribute::Class(class);
    let publickeyhandle =  session.find_objects(&[pub_template]).unwrap()[0];
    let new_pub_key = session.get_attributes(publickeyhandle, &[AttributeType::Value]).unwrap();
}
```

Now how do I use `new_pub_key` to generate the csr using openssl

**Comment by wiktor-k** [comment, marked as answer]
Maybe this snippet taken from tests would help?

https://github.com/parallaxsecond/rust-cryptoki/blob/main/cryptoki/tests/basic.rs#L184-L191

**Reply by shivam560** [reply]
Thanks It worked, I dont know how missed that snippet but Thanks for the help. :)
